packages/core-data/src/hooks/use-resource-permissions.ts

```diff
@@ -109,15 +124,36 @@ type ResourcePermissionsResolution< IdType > = [
  * @return Entity records data.
  * @template IdType
  */
-export default function useResourcePermissions< IdType = void >(
-	resource: string,
+function useResourcePermissions< IdType = void >(
+	resource: string | EntityResource,
 	id?: IdType
 ): ResourcePermissionsResolution< IdType > {
+	// Serialize `resource` to a string that can be safely used as a React dep.
+	// We can't just pass `resource` as one of the deps, because if it is passed
+	// as an object literal, then it will be a different object on each call even
+	// if the values remain the same.
+	const resourceAsString =
+		typeof resource === 'object' ? JSON.stringify( resource ) : resource;
+
+	const isEntity = typeof resource === 'object';
```

Comment:
We can just reuse the `isEntity` constant:

```suggestion
	const isEntity = typeof resource === 'object';
	const resourceAsString = isEntity ? JSON.stringify( resource ) : resource;
```